pkg/refactor/logstore/dblogstore/logstore.go

```diff
@@ -0,0 +1,279 @@
+package dblogstore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/direktiv/direktiv/pkg/refactor/logstore"
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+const (
+	ns  = "namespace"
+	wf  = "workflow"
+	srv = "server"
+	ins = "instance"
+	mir = "mirror"
+)
+
+var _ logstore.LogStore = &SQLLogStore{} // Ensures SQLLogStore struct conforms to logstore.LogStore interface.
+
+func NewSQLLogStore(db *gorm.DB) logstore.LogStore {
+	return &SQLLogStore{
+		db: db,
+	}
+}
+
+type SQLLogStore struct {
+	db *gorm.DB
+}
+
+// Append implements logstore.LogStore.
+// - For instance-logs following Key Value pairs SHOULD be present: instance-id, callpath, root-instance-id
+// - For namespace-logs following Key Value pairs SHOULD be present: namespace-id
+// - For mirror-logs following Key Value pairs SHOULD be present: mirror-id
+// - For workflow-logs following Key Value pairs SHOULD be present: workflow-id
+// - All passed keysAndValues pair will be stored attached to the log-entry.
+func (sl *SQLLogStore) Append(ctx context.Context, timestamp time.Time, msg string, keysAndValues ...interface{}) error {
+	fields, err := mapKeysAndValues(keysAndValues...)
+	cols := make([]string, 0, len(keysAndValues))
+	vals := make([]interface{}, 0, len(keysAndValues))
+	if err != nil {
+		return err
+	}
+	cols = append(cols, "t", "msg")
+	vals = append(vals, timestamp, msg)
+	lvl, ok := fields["level"]
+	if ok {
+		switch lvl {
+		case "debug", "info", "error", "panic":
```

Comment:
This store is designed to be very stupid that only maps golang function call to a sql query. Any validation/processing of the data should be implemented in the higher level.